operate/schema/src/main/java/io/camunda/operate/schema/IndexSchemaValidator.java

```diff
@@ -35,22 +35,10 @@ public class IndexSchemaValidator {
 
   @Autowired SchemaManager schemaManager;
 
-  @Autowired private OperateProperties operateProperties;
-
-  private Set<String> getAllIndexNamesForIndex(final String index) {
-    final String indexPattern = String.format("%s-%s*", getIndexPrefix(), index);
+  private Set<String> getAllIndexNamesForIndex(final IndexDescriptor index) {
+    final String indexPattern = index.getAllVersionsIndexNameRegexPattern();
     LOGGER.debug("Getting all indices for {}", indexPattern);
-    final Set<String> indexNames = schemaManager.getIndexNames(indexPattern);
-    // since we have indices with similar names, we need to additionally filter index names
-    // e.g. task and task-variable
-    final String patternWithVersion = String.format("%s-%s-\\d.*", getIndexPrefix(), index);
```

Comment:
this regex is already present when using `index.getAllVersionsIndexNameRegexPattern();`